docs/developer-guide/rules/index.md

```diff
@@ -11,12 +11,113 @@ flexible enough to allow you to implement anything you want easily:
 If there is something you want to do and you can't, or it is not clear
 how to do it, please open an issue.
 
-The following is a basic template for a core rule (you might need to
-adapt the paths for your case):
+## Types of rules
+
+There are 2 types of `rule`s a user can develop:
+
+* `external`: These are `rule`s that are published independently. When a
+  `rule` is specific of a domain or use case, it should be external.
+* `core`: These are the `rule`s that are shipped with `sonar` directly.
+  Before starting to develop a `core rule`, please make sure there is
+  an open issue and talk with the maintainers about it.
+
+Both types of `rule`s [work exactly the same](#howruleswork). It is just
+where they are located that changes.
+
+### Creating an external rule
+
+To create a new rule that will be distributed outside `sonar` you will
```

Comment:
Maybe use:

> The easiest way to create a new rule...

As users can do it manually if they want to, or what to do it in pure JavaScript. 

---

We should look into suggesting [npx](http://blog.npmjs.org/post/162869356040/introducing-npx-an-npm-package-runner).

Reply:
> We should look into suggesting npx.

Added, unfortunately [Windows users have problems])(https://github.com/npm/npm/issues/17869) with this option :(